test/src/nonunity/userobjects/TestNEML2StressUO.C

```diff
@@ -0,0 +1,129 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifdef NEML2_ENABLED
+
+#include "neml2/base/HITParser.h"
+#include "TestNEML2StressUO.h"
+#include "MathUtils.h"
+
+registerMooseObject("BlackBearTestApp", TestNEML2StressUO);
+
+InputParameters
+TestNEML2StressUO::validParams()
+{
+  auto params = TestNEML2StressUOParent::validParams();
+  params.addRequiredParam<MaterialPropertyName>("strain", "The name of strain");
```

Comment:
Shouldn't we default to `mechanical_strain`?  I guess you have in mind later models that might want to latch onto something else?  Except we are tied to the kernels -- we have to return d_stress / d_mechanical_strain for small deformations and d_cauchy_stress / d_spatial_velocity_increment for large strains.  So can the user really choose the name of the strain?